packages/cli/docs/custom-plugins.md

```diff
@@ -0,0 +1,637 @@
+      <th>Severity</th>
+      <th>Message</th>
+      <th>Source file</th>
+      <th>Line(s)</th>
+    </tr>
+    <tr>
+      <td>🚨 <i>error</i></td>
+      <td>File file-a.js is 17.31 kB bytes too big. ( budget: 41.02 kB)</td>
+      <td><code>/src/file-a.js</code></td>
+      <td></td>
+    </tr>
+    <tr>
+      <td>ℹ️ <i>info</i></td>
+      <td>File file-b.js OK</td>
+      <td><code>/src/file-b.js</code></td>
+      <td></td>
+    </tr>
+  </table>
+</details>
+```
+
+## Plugins and categories
+
+@TODO An intro text is missing here. What is this section trying to convey?
+
+**lighthouse plugin**
+
+```typescript
+// code-pushup.config.ts
+import { CoreConfig } from '@code-pushup/core';
+
+export default {
+  // ...
+  categories: [
+    {
+      slug: 'performance',
+      title: 'Performance',
+      description: "These encapsulate your app's current performance and opportunities to improve it.",
+      refs: [
+        {
+          type: 'audit',
+          slug: 'largest-contentful-paint',
+          plugin: 'lightouse',
+          weight: 1,
+        },
+      ],
+    },
+  ],
+} satisfies CoreConfig;
+```
+
+Test the output by running `npx code-pushup collect --no-progress`.
+
+**stdout of basic lighthouse plugin**
+
+```sh
+Code PushUp Report - @code-pushup/core@x.y.z
+
+Chrome Lighthouse audits
+● Largest Contentful Paint                                                1,3 s
+
+Categories
+┌─────────────┬───────┬────────┐
+│ Category    │ Score │ Audits │
+├─────────────┼───────┼────────┤
+│ Performance │    42 │      1 │
+└─────────────┴───────┴────────┘
+
+
+Made with ❤ by code-pushup.dev
+```
+
+### Score Audits
+
+Every audit has a score as floating number between 0 and 1.
+We will extend the file-size example to calculate the score based on a budget.
+
+Let's first extend the options object with a `budget` property and use it in the runner config:
+
+**file-size plugin form section [RunnerFunction](#RunnerFunction)**
+
+```typescript
+// code-pushup.config.ts
+type Options = {
+  // ...
+  budget: number; // in bytes
+};
+
+export default {
+  // ...
+  plugins: [
+    await createPlugin({
+      directory: './src',
+      budget: 1000,
+    }),
+  ],
+  // ...
+};
+```
+
+Now let's extend the runner function to calculate the score based on the budget:
+
+**file-size plugin runnerFunction**
+
+```typescript
+// code-pushup.config.ts
+
+// ...
+
+async function runnerFunction(options: Options): Promise<AuditOutputs> {
+  // ...
+
+  // score audit based on budget
+  const errorCount = data.map(({ size }) => options.budget < size).filter(Boolean).length;
+  if (errorCount) {
+    fileSizeAuditOutput: AuditOutput = {
+      ...fileSizeAuditOutput,
+      // score is percentage of over-budget files
+      score: errorCount ? errorCount / issues.length : 1,
+      value: errorCount,
+      displayValue: `${errorCount} ${errorCount === 1 ? 'file' : pluralize('file')}`,
+    };
+  }
+
+  return [fileSizeAuditOutput];
+}
+```
+
```

Comment:
It would be nice to reuse the MD example with an actual audit (or merged with group) score value and add a note above the report stating where the value is artificial and when it is real.